src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -122,6 +123,7 @@ static ProjectCodeRefactoringProvider.ExtensionInfo GetExtensionInfo(ExportCodeR
             Func<string, IDisposable?> addOperationScope,
             CancellationToken cancellationToken)
         {
+            using (TelemetryLogging.LogBlockTimeAggregated(FunctionId.CodeRefactoring_Summary, "Pri" + (int)priority))
```

Comment:
So, I like the fact that you have added lot of different fine grained aggregated log blocks for different parameter combinations (such as specific priority, fix vs refactoring, specific provider, etc.). The tricky part for someone new looking at this telemetry might be to figure out which ones to dig into for investigating their specific scenario. I think eventually adding a document or even a brief summary somewhere about the different type of events logged in each space (diagnostics, quick actions, etc.) with a small description will be helpful. Basically, documentation from investigation perspective, not the actual implementation perspective.

Reply:
I don't disagree, but it seems to me that such documentation would be very easy to get out of date. I think the dashboards are a nice piece of "documentation" from this perspective as they give the actual rundowns of the telemetry events that exist and are being fired.